PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h

```diff
@@ -271,3 +272,134 @@ class FirstObjectSimpleFlatTableProducer : public SimpleFlatTableProducerBase<T,
     return out;
   }
 };
+
+template <typename T, typename TProd>
+class SimpleFlatTableProducerBaseLumi : public edm::one::EDProducer<edm::EndLuminosityBlockProducer> {
+public:
+  SimpleFlatTableProducerBaseLumi(edm::ParameterSet const &params)
+      : name_(params.getParameter<std::string>("name")),
+        doc_(params.existsAs<std::string>("doc") ? params.getParameter<std::string>("doc") : ""),
+        extension_(params.existsAs<bool>("extension") ? params.getParameter<bool>("extension") : false),
+        skipNonExistingSrc_(
+            params.existsAs<bool>("skipNonExistingSrc") ? params.getParameter<bool>("skipNonExistingSrc") : false),
+        src_(skipNonExistingSrc_ ? mayConsume<TProd>(params.getParameter<edm::InputTag>("src"))
```

Comment:
Why `mayConsume`?

Reply:
Dear @makortel,

Thank you very much for your comments.

I based the 'Lumi version' of the module off the existing _SimpleFlatTableProducerBase_: https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h#L23

Therefore, I think this question is best addressed to the original developers. If it were to be updated, then I think it would make sense that it is updated consistently in both modules by the original developers.

Best,
Mateusz